force-app/main/default/classes/HouseholdMembers.cls

```diff
@@ -0,0 +1,243 @@
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @description Domain class for HouseholdMembers in NPSP.
+*/
+public inherited sharing class HouseholdMembers implements IHouseholdMembers {
+
+    List<Contact> records;
+    public HouseholdMembers(List<Contact> records) {
+        this.records = records;
+    }
+
+    public Map<Id, Contact> oldMap;
+    public HouseholdMembers(List<Contact> records, Map<Id, Contact> oldMap) {
+        this.records = records;
+        this.oldMap = oldMap != null ?
+                new Map<Id, Contact>(oldMap) : new Map<Id, Contact>();
+    }
+
+    public List<Contact> forHouseholdName() {
+        List<Contact> contactsForHouseholdNameGeneration = new List<Contact>();
+        for (Contact con : records) {
+            if (isIncludedInHouseholdName(con)) {
+                contactsForHouseholdNameGeneration.add(con);
+            }
+        }
+        return contactsForHouseholdNameGeneration;
+    }
+
+    public List<Contact> forFormalGreeting() {
+        List<Contact> contactsForFormalName = new List<Contact>();
+        for (Contact con : records) {
+            if (isIncludedInFormalGreeting(con)) {
+                contactsForFormalName.add(con);
+            }
+        }
+        return contactsForFormalName;
+    }
+
+    public List<Contact> forInformalGreeting() {
+        List<Contact> listCon = new List<Contact>();
+        for (Contact con : records) {
+            if (isIncludedInInformalGreeting(con)) {
+                listCon.add(con);
+            }
+        }
+        return listCon;
+    }
+
+    private Boolean isIncludedInHouseholdName(Contact con) {
+        HouseholdNamingExclusionsString exclusionsAsString = new
+                HouseholdNamingExclusionsString(con);
+        return !exclusionsAsString.isExcludedFromName();
+    }
+
+    private Boolean isIncludedInFormalGreeting(Contact con) {
+        HouseholdNamingExclusionsString exclusionsAsString = new
+                HouseholdNamingExclusionsString(con);
+        return !exclusionsAsString.isExcludedFromFormalGreeting();
+    }
+
+    private Boolean isIncludedInInformalGreeting(Contact con) {
+        HouseholdNamingExclusionsString exclusionsAsString = new
+                HouseholdNamingExclusionsString(con);
+        return !exclusionsAsString.isExcludedFromInformalGreeting();
+    }
+
+    public void createAddressAfterInsertUpdate(TDTM_Runnable.DmlWrapper dmlWrapper) {
```

Comment:
I think the `AfterInsertUpdate` can be removed from the method name since it is up to trigger/adapter to determine when to call the method. Also there is no similar method name in the class that needs the trigger context to separate the difference